node/server.js

```diff
@@ -5585,6 +5585,50 @@ function init_io() {
 			resend(player, "reopen+nc+inv");
 			success_response({ num: num, cevent: true });
 		});
+
+		socket.on("destat", function (data) {
+			var player = players[socket.id];
+			var item = player.items[data.item_num];
+			var def = G.items[item && item.name];
+			if (!player || player.user) {
+				return fail_response("cant_in_bank");
+			}
+			if (!player.computer && simple_distance(G.maps.desertland.ref.scrollsmith, player) > B.sell_dist) {
+				return fail_response("distance");
+			}
+			var item = player.items[data.num];
+			if (!item) {
+				return fail_response("no_item");
+			}
+			var def = G.items[item.name];
+			if (item.stat_type == null) {
+				return fail_response("scrollsmith_cant");
+			}
+			var scrolltype = item.stat_type + "scroll";
+			if (scrolltype == "mp_costscroll") {
+				scrolltype = "mpcostscroll";
```

Comment:
Did you try with all scrolls to make sure they all follow the pattern except `mpcostscroll`?